src/vllm_haystack/__init__.py

```diff
@@ -322,35 +304,26 @@ def invoke(self, *args, **kwargs):
 
         :return: The responses are being returned.
 
-        Note: Only kwargs relevant to OpenAI are passed to OpenAI rest API. Others kwargs are ignored.
-        For more details, see OpenAI [documentation](https://platform.openai.com/docs/api-reference/completions/create).
+        Note: Only kwargs relevant to vLLM are passed to local vLLM model. Others kwargs are ignored.
         """
         prompt = kwargs.get("prompt")
         # either stream is True (will use default handler) or stream_handler is provided
-        kwargs_with_defaults = self.model_input_kwargs
+        kwargs_with_defaults = self.model_input_kwargs.copy()
+
         if kwargs:
             # we use keyword stop_words but OpenAI uses stop
             if "stop_words" in kwargs:
-                kwargs["stop"] = kwargs.pop("stop_words")
+                kwargs_with_defaults["stop"] = kwargs.pop("stop_words")
             if "top_k" in kwargs:
                 top_k = kwargs.pop("top_k")
-                kwargs["n"] = top_k
-                kwargs["best_of"] = top_k
-            kwargs_with_defaults.update(kwargs)
+                kwargs_with_defaults["n"] = top_k
+                kwargs_with_defaults["best_of"] = top_k
+            for k, v in kwargs.items():
+                if k in self._allowed_params:
+                    kwargs_with_defaults[k] = v 
+        _stream = (kwargs.get("stream", False) or kwargs.get("stream_handler", None) is not None) or self._stream
```

Comment:
By shenanigans do you mean renaming stop_words and reusing top_k? I agree it should be explicit, but I don't know why it was the way it was so I kept it.

One thing I think should be fixed about this is to raise an error if unrecognized kwargs are passed in rather than just accepting and ignoring them

Reply:
> One thing I think should be fixed about this is to raise an error if unrecognized kwargs are passed in rather than just accepting and ignoring them

I think most of haystacks components silently accept unrecognized kwargs. I based the vLLMInvokation layer on their existing openai layer, thats why it inherited most of the kwargs.